### PR TITLE
AO3-6762: Restrict access to actions on TagsController

### PR DESCRIPTION
**Pull request by Cesium-Ice**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6762

## Purpose

What does this PR do?

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended?

If you have a Jira account with access, please update or comment on the issue
with any new or missing testing instructions instead.

## References

Are there other relevant issues/pull requests/mailing list discussions?

## Credit

What name and pronouns should we use to credit you in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?

Cesium-Ice, they/them

If you have a Jira account, please include the same name in the "Full name"
field on your Jira profile, so we can assign you the issues you're working on.

*Please note that if you do not fill in this section, we will use your GitHub account name and
they/them pronouns.*

### CODE REVIEW

File: app/policies/wrangling_policy.rb [hunk 1]
```diff
@@ -2,14 +2,14 @@
 
 class WranglingPolicy < ApplicationPolicy
   FULL_ACCESS_ROLES = %w[superadmin tag_wrangling].freeze
-  VIEW_ACCESS_ROLES = (FULL_ACCESS_ROLES + %w[policy_and_abuse]).freeze
+  READ_ACCESS_ROLES  = (FULL_ACCESS_ROLES + %w[policy_and_abuse]).freeze
```
Comment (brianjaustin) [inline app/policies/wrangling_policy.rb:5]: Please apply the Rubocop suggestion, just a quick space removal:
```suggestion
  READ_ACCESS_ROLES = (FULL_ACCESS_ROLES + %w[policy_and_abuse]).freeze
```
(for files that don't have other violations you can also do `rubocop -a <path/to/file>`, but that gets tricky with some old files that need loads of fixes)

File: spec/controllers/tags_controller_spec.rb [hunk 2]
```diff
@@ -464,45 +374,14 @@
...
-          put :update, params: { id: tag, tag: { syn_string: synonym.name }, commit: "Save changes" }
-          it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
-        end
+      subject { put :update, params: { id: tag, tag: { syn_string: synonym.name }, commit: "Save changes" } }
+      let(:success) do
+        it_redirects_to_with_notice(edit_tag_path(tag), "Tag was updated.")
+        tag.reload
+        expect(tag.merger_id).to eq(synonym.id)
       end
+      it_behaves_like "an action only fully authorized admins can access", authorized_roles: wrangling_full_access_roles
```
Comment (brianjaustin) [inline spec/controllers/tags_controller_spec.rb:384]: Small formatting nitpick:
```suggestion

      it_behaves_like "an action only fully authorized admins can access", authorized_roles: wrangling_full_access_roles
```

File: spec/controllers/tags_controller_spec.rb [hunk 3]
```diff
@@ -17,6 +17,41 @@
     end
   end
 
+  shared_examples "an action only fully authorized admins can access" do |authorized_roles:|
```
Comment (brianjaustin) [inline spec/controllers/tags_controller_spec.rb:20]: To me, "fully" implies the admin has write access. How about this?
```suggestion
  shared_examples "an action only authorized admins can access" do |authorized_roles:|
```

**Review by brianjaustin — CHANGES_REQUESTED** [review]
Thank you! Just a few very small things, please let me know if you have any questions about them

**Review by brianjaustin — APPROVED** [review]
Thank you!

**Comment by Bilka2** [comment]
This branch has a merge conflict, could you merge master and solve the conflict? We'd like to include this PR in a release soon.

**Pull request merged**